src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests_NameDeclarationInfoTests.cs

```diff
@@ -162,6 +162,60 @@ void goo()
             await VerifyAccessibility(markup, Accessibility.NotApplicable);
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void ForVariableDeclaration1()
+        {
+            var markup = @"
+class C
+{
+    void M()
+    {
+        for (int i$$
+    }
+}
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+            await VerifyModifiers(markup, new DeclarationModifiers());
+            await VerifyTypeName(markup, "int");
+            await VerifyAccessibility(markup, Accessibility.NotApplicable);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void ForVariableDeclaration2()
+        {
+            var markup = @"
+class C
+{
+    void M()
+    {
+        for (int i1, $$
+    }
+}
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+            await VerifyModifiers(markup, new DeclarationModifiers());
+            await VerifyTypeName(markup, "int");
+            await VerifyAccessibility(markup, Accessibility.NotApplicable);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void ForEachVariableDeclaration()
+        {
+            var markup = @"
+class C
+{
+    void M()
+    {
+        foreach (int $$
+    }
+}
+";
+            await VerifySymbolKinds(markup, SymbolKind.Local);
+            await VerifyModifiers(markup, new DeclarationModifiers());
+            await VerifyTypeName(markup, "int");
+            await VerifyAccessibility(markup, Accessibility.NotApplicable);
+        }
```

Comment:
Thanks for fixing this!
If you want to take a look, there are other ways to declare locals which might be affected by the same bug (name completion not triggering): `out` variables, pattern variables, `using`, deconstruction declaration, deconstruction foreach.

Reply:
> out variables, pattern variables

Out & pattern variables work.
> using

You're right about using, I hadn't thought of that.

> deconstruction declaration, deconstruction foreach

Deconstructions are not supported at all right now. Neither are tuples. I might do that later, but this was just a quick PR to fix something I ran into and was bothering me.

Reply:
@jcouv 
> out variables, pattern variables

Out & pattern variables work.
> using

You're right about using, I hadn't thought of that.

> deconstruction declaration, deconstruction foreach

Deconstructions are not supported at all right now. Neither are tuples. I might do that later, but this was just a quick PR to fix something I ran into and was bothering me.

Reply:
It seems there's a PR for tuples already: #22673

Reply:
Support for using added.